### DISCUSSION TITLE
How to get Elasticache: Quota Limits with cli?

### DISCUSSION THREAD

**Discussion by vee-16** [aws/aws-cli#8310] [Q&A] [answered]
Hi all,

Is there a way to get/view ElastiCache service quota limits without using the the [Service Quotas console](https://console.aws.amazon.com/servicequotas/home)? 

- I've been using the CLI(tried with version```1.29.81```) to get:

```aws service-quotas list-service-quotas --service-code elasticache --region your-region```

and see: ```Quotas: []```

- However, with ```--service-code ec2```:

```aws service-quotas list-service-quotas --service-code ec2 --region your-region```, I am seeing results back.

Seem to be there is no current support for ```--service-code elasticache``` while searching through documentations. 
I've also tried using ```github.com/aws/aws-sdk-go-v2/service/servicequota``` but no luck.

Is there something I have missed? It would be greatly beneficial to view our limit and current applied value to determine if we are approaching a quota limit before we request an increase.

Thank you for having a look.

**Comment by RyanFitzSimmonsAK** [comment, marked as answer]
Hi @vee-16, thanks for reaching out.  ListServiceQuotas has varying levels of support on a service-by-service basis.  In the [documentation](https://docs.aws.amazon.com/servicequotas/2019-06-24/apireference/API_ListServiceQuotas.html), it says "For some quotas, only the default values are available. If the applied quota value is not available for a quota, the quota is not retrieved.".

`list-aws-default-service-quotas` works for all services, but you mentioned that you have applied quotas, so if that isn't sufficient for your use case, I can reach out to the Elasticache service team for you.  Just let me know how you'd like to proceed here.  Thanks!

@github-actions proposed answer

**Reply by vee-16** [reply]
Hi @RyanFitzSimmonsAK,

Thanks for taking the time to help with my question!

Yes, could you please contact the ElastiCache service team about this issue and inquire if they recommend a workaround solution?

To provide a bit more context for the user story we're developing, our focus areas are:

* 		View current applied quota limits
* 		View current utilization of resource
* 		Utilize the AWS CLI for requesting quota increases from the support team

This information will help us monitor and alert for ElastiCache resources which is important to prevent service disruptions and ensure optimal resource utilization.

Let me know if I can help clarify things further.

**Reply by vee-16** [reply]
Hey @RyanFitzSimmonsAK,

any updates on this issue? Thank you for taking a look!

**Reply by RyanFitzSimmonsAK** [reply]
Hey, thanks for your patience.  The Elasticache team is already tracking an existing request for this feature, but don't have a specific timeline that can be shared.  Unfortunately, there aren't any existing workarounds for checking applied quotas programmatically, so you'll probably need to contact AWS Support to do that.
